parity/configuration.rs

```diff
@@ -736,11 +736,10 @@ impl Configuration {
 		let local_path = default_local_path();
 		let data_path = replace_home("", self.args.flag_datadir.as_ref().unwrap_or(&self.args.flag_base_path));
 
-		let db_path = if self.args.flag_datadir.is_some() {
-			replace_home(&data_path, &self.args.flag_db_path)
-		} else {
-			replace_home_for_db(&data_path, &local_path, &self.args.flag_db_path)
-		};
+		//TODO: refactor this for better detection if base_path is custom.
+		let base_db_path = if data_path != default_data_path() && self.args.flag_db_path == dir::CHAINS_PATH { "$BASE/chains".to_owned() } else { self.args.flag_db_path.clone() };
```

Comment:
`data_path != default_data_path()` looks pretty error prone... if it's a quick hack, then we should probably have an issue logged to fix it into something sensible asap.